### DISCUSSION TITLE
Consider un-deprecating Subject.isStopped

### DISCUSSION THREAD

**Discussion by johncrim** [ReactiveX/rxjs#6462] [Ideas / Feature request]
Since a Subject is the root multicast object in rx.js, it's useful to know whether it has completed or errored, and thus whether sending more items or an error to the Subject is going to do anything.

With `Subject.isStopped` now being deprecated in 7.x, it's become more effort to track the state of the Subject. Could you provide a getter with equivalent information?

The best alternative I see is adding Subject/BehaviorSubject subclasses, or a wrapper that tracks the same, but I figure it's worth asking for.

Thanks in advance!

**Comment by raymonddavis** [comment]
Can you use closed?
https://github.com/ReactiveX/rxjs/blob/f47425d349475231c0f3542bb6ecef16a63e933a/src/internal/Subject.ts#L17

**Reply by ncjamieson** [reply]
Nope. `closed` will only be `true` when `unsubscribe` is called _directly upon_ the `Subject`:

https://github.com/ReactiveX/rxjs/blob/4f4ac0074b4bc8aead728e2fa7aa0a9061f56f0b/src/internal/Subject.ts#L89-L92

`isStopped` was deprecated because it's a leaked implementation detail. It was never intended to be part of the public API.

**Reply by raymonddavis** [reply]
@ncjamieson ah I see. I missed it getting set in error and complete.

**Comment by johncrim** [comment]
The thing that `isStopped` gives me, which I can't get otherwise (trivially) is "this subject has been completed or errored, so will ignore any further input or errors". I acknowledge this behavior is the same as "no one's listening", but the `Subject.observed` property gives me that info.

I think `isStopped` is useful for `Subject`, and unnecessary for `Observable`, b/c in the single observable -> single observer case there's a new process for each `subscribe()` so the observable has to track what it's sent; whereas Subject is stateful and can be used for multiple senders/multiple receivers, so it's useful to be able to check this state on the `Subject`.

**Comment by backbone87** [comment]
if that is introduced, it should be similar how the `share` operator tracks this internally (so it can be reused by share et al):
```ts
interface Subject {
  completed: boolean; // if true, broadcasting through this subject leads to stopped notifications
  errored: boolean; // if true, broadcasting through this subject leads to stopped notifications
  closed: boolean; // if true, broadcasting through this subject causes an error to be thrown
  observed: boolean; // if true, someone will actually listen to broadcasts (unless we have special subjects)
}
```
